pgmpy/models/BayesianModel.py

```diff
@@ -316,20 +316,50 @@ def remove_cpds(self, *cpds):
                 cpd = self.get_cpds(cpd)
             self.cpds.remove(cpd)
 
-    def get_cardinality(self, node):
+    def get_cardinality(self, node=None):
         """
         Returns the cardinality of the node. Throws an error if the CPD for the
         queried node hasn't been added to the network.
 
         Parameters
         ----------
-        node: Any hashable python object.
+        node: Any hashable python object(optional).
+              The node whose cardinality we want. If node is not specified returns a
+              dictionary with the given variable as keys and their respective cardinality
+              as values.
 
         Returns
         -------
-        int: The cardinality of the node.
+        int: If node is specified returns the cardinality of the node.
```

Comment:
@lohani2280 In my opinion having something like `int or dict: The description with the conditions` is better because have two separate `int` and `dict` points might suggest multiple return values.

Reply:
@ankurankan Yes, it would be better . I will update the pr soon.